packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -1510,24 +1573,38 @@ private Boolean connectToDeprecated(
     if (!enabled) return false;
 
     boolean connected = false;
+    int networkId = -1;
     for (int i = 0; i < 20; i++) {
       WifiInfo currentNet = moWiFi.getConnectionInfo();
-      int networkId = currentNet.getNetworkId();
+      networkId = currentNet.getNetworkId();
       SupplicantState netState = currentNet.getSupplicantState();
 
       // Wait for connection to reach state completed
       // to discard false positives like auth error
       if (networkId != -1 && netState == SupplicantState.COMPLETED) {
         connected = networkId == updateNetwork;
-        break;
+        if(connected == true) {
+          break;
+        } else {
+          disconnect = moWiFi.disconnect();
+          if (!disconnect) {
+            break;
+          }
+
+          enabled = moWiFi.enableNetwork(updateNetwork, true);
```

Comment:
Looks like this updated `enabled` value isn't getting used after updating it here. Is it even necessary to assign it to a value if it won't get read after this line?

Reply:
@KhatibFX Can you explain this change block? why is it necessary?

Reply:
> https://github.com/flutternetwork/WiFiFlutter/blob/796f784fd1d3554bf80612bea90b577ecace8c9a/packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java#L1586-L1596
> 
> @KhatibFX Can you explain this change block? why is it necessary?

So after `netState == SupplicantState.COMPLETED` - we check if connected, if not then disconnect and "enable network". I am not sure what the reason behind it.

Also, can the code be simplified? - avoiding `break` in each if-else branch.

Reply:
So the reason here is that sometimes after supplicantstate.completed, it turns out the connection failed. On newer android versions, the phone will automatically revert back to the WiFi connection that was being used before trying to connect to this new network. On older Android versions, it will just remain disconnected and won't attempt to reconnect to the previously connected network. This part of the code attempts to do that reconnection step, to make the experience similar to newer Android versions. From my testing, it wasn't enough to just ask to enableNetwork of the previous network. I had to fully reset the connection by disconnecting and then reenabling the created network.

There may be a better way to rewrite this block better, but I am not amazing in Java so I'm not sure.